cleanlab_studio/utils/synthetic.py

```diff
@@ -0,0 +1,219 @@
+        High values indicate there exist tails
+        of the real data distribution (rare scenarios)
+        that the distribution of synthetic samples fails to capture.
+        """
+
+        return self._lis_to_synthetic_score(self.real_type)
+
+    def _unvaried_score(self) -> float:
+        """Evaluates the diversity among synthetic samples.
+        High values indicate a non-diverse synthetic data generator
+        that produces many similar samples which look like near duplicates
+        of other synthetic samples.
+        """
+        target_type = self.synthetic_type
+        return self._calculate_synthetic_duplicate_scores(target_type, False)
+
+    def _unoriginal_score(self) -> float:
+        """Evaluates the lack of novelty of the synthetic data.
+        High values indicate many synthetic samples are near duplicates
+        of an example from the real dataset,
+        i.e. the synthetic data generator may be memorizing the real data
+        too closely and failing to generalize.
+        """
+        return self._calculate_synthetic_duplicate_scores(
+            self.synthetic_type,
+            True,
+        )
+
+    def _lis_to_synthetic_score(self, target_type: str) -> float:
+        """Computes the synthetic score based on
+        the label quality scores of the dataset.
+
+        Parameters
+        ----------
+        target_type: str
+            The type of examples to compute the score for. If the
+            target type is "synthetic", or a class name that
+            corresponds to synthetic examples, then the score is
+            computed based on the label issue scores of the
+            synthetic examples.
+            If the target type is "real",
+            or a class name that corresponds to real examples,
+            then the score is computed based on the label
+            issue scores of the real examples.
+
+        Returns
+        -------
+        score: float
+            The synthetic score, which is the mean label issue score
+            of the target type.
+        """
+        query = f"{self.real_or_synth_column} == @target_type"
+        filtered_df = self.cleanset_df.query(query)
+        mean_label_issue_score = filtered_df["label_issue_score"].mean()
+        score = 1 - mean_label_issue_score
+        return cast(float, score)
+
+    def _calculate_synthetic_duplicate_scores(
+        self,
+        target_type: str,
+        contains_real: bool,
+    ) -> float:
+        # Real or synthetic column
+        _rs_column = self.real_or_synth_column
+
+        # Synthetic class names (real or synthetic)
+        class_names = self.cleanset_df[_rs_column].unique()
+        assert len(class_names) == 2, "The dataset should contain both real and synthetic examples"
+        target_mask = class_names == target_type
+        complementary_class: str = class_names[~target_mask][0]
+
+        df_near_duplicates = self.cleanset_df.query("is_near_duplicate")
```

Comment:
wondering why we make `near_duplicate_id_column` a field on `_SyntheticDatasetScorer` but not this "is_near_duplicate" column name

Reply:
Well, `near_duplicate_id_column` may as well be a constant defined at the top of the module. I didn't see a reason to make one for `"is_near_duplicate"`, as it's only used once in this module.

Reply:
Personally would prefer making `near_duplicate_id_column` a constant since it shouldn't change between instances of `_SyntheticDatasetScorer`